dnssec.go

```diff
@@ -543,35 +545,37 @@ func (k *DNSKEY) publicKeyRSA() *rsa.PublicKey {
 		keyoff = 3
 	}
 
-	if explen > 4 || explen == 0 || keybuf[keyoff] == 0 {
+	pubkey := new(rsa.PublicKey)
+	maxexplen := uint16(strconv.IntSize / 8)
```

Comment:
is this the official way to do this, looks very iffy to me ....  ?
why should we check this in the first place if the crypto package handles it for us? Can't we just create a key and see what happens?

Reply:
The whole function looks quite "iffy" to me ;-) I've tried to the minimal change to get it working. The recommended refactoring would be a different PR in my opinion.